src/main/java/com/liferay/faces/util/cache/CacheFactory.java

```diff
@@ -0,0 +1,228 @@
+/**
+ * Copyright (c) 2000-2017 Liferay, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.liferay.faces.util.cache;
+
+import javax.faces.context.ExternalContext;
+
+import com.liferay.faces.util.factory.FactoryExtensionFinder;
+import com.liferay.faces.util.helper.Wrapper;
+
+
+/**
+ * @author  Kyle Stiemann
+ * @since   3.1
+ * @since   2.1
+ * @since   1.1
+ */
+public abstract class CacheFactory implements Wrapper<CacheFactory> {
+
+	/**
+	 * Returns a new instance of {@link Cache} from the {@link CacheFactory} found by the {@link
+	 * FactoryExtensionFinder}. The returned instance is designed to be used during execution of a request thread, so it
+	 * is not guaranteed to be {@link java.io.Serializable}.
+	 *
+	 * @param  <K>              The type of the cache's keys.
+	 * @param  <V>              The type of the cache's values.
+	 * @param  externalContext  The external context associated with the current faces context. It is needed in order
+	 *                          for the {@link FactoryExtensionFinder} to be able to find the factory.
+	 */
+	public static <K, V> Cache<K, V> getCacheInstance(ExternalContext externalContext) {
+
+		CacheFactory cacheFactory = (CacheFactory) FactoryExtensionFinder.getFactory(externalContext,
+				CacheFactory.class);
+
+		return cacheFactory.<K, V>getCache();
+	}
+
+	/**
+	 * Returns a new instance of {@link Cache} from the {@link CacheFactory} found by the {@link
+	 * FactoryExtensionFinder}. The returned instance is designed to be used during execution of a request thread, so it
+	 * is not guaranteed to be {@link java.io.Serializable}.
+	 *
+	 * @param  <K>              The type of the cache's keys.
+	 * @param  <V>              The type of the cache's values.
+	 * @param  externalContext  The external context associated with the current faces context. It is needed in order
+	 *                          for the {@link FactoryExtensionFinder} to be able to find the factory.
+	 * @param  initialCapacity  The initial capacity of the cache.
+	 */
+	public static <K, V> Cache<K, V> getCacheInstance(ExternalContext externalContext, int initialCapacity) {
```

Comment:
@stiemannkj1 I've noticed that JavaDoc like [PortletResponse.addProperty(Cookie)](https://portals.apache.org/pluto/portlet-3.0-apidocs/javax/portlet/PortletResponse.html#addProperty(javax.servlet.http.Cookie)) documents when a method throws `IllegalArgumentException`. I recommend that we do the same in all CacheFactory methods here that take an `initialCapacity` or `maxCapacity` argument.